client/client.go

```diff
@@ -214,6 +188,36 @@ func (c *LogClient) UpdateRoot(ctx context.Context) error {
 	return nil
 }
 
+// WaitForInclusion blocks until the requested data has been verified with an inclusion proof.
```

Comment:
Might be worth adding something like "This obviously assumes that the data has already been submitted (e.g. with QueueLeaf)." to the doc comment just to be clear.